### PR TITLE
Modify the priority of :any routes, fixes #1089

### PR DESCRIPTION
**Pull request by namusyaka**
ref #1089 

Conventional, the :any routes had been searched after matching first route and 405 routes.
This PR fixes the issue by adding the :any routes processing before 405 processing.

### CODE REVIEW

File: spec/grape/api_spec.rb [hunk 1]
```diff
@@ -755,9 +786,6 @@ class DummyFormatClass
         subject.post :example do
           'example'
         end
-        subject.route :any, '*path' do
-          error! :not_found, 404
-        end
```
Comment (namusyaka) [inline spec/grape/api_spec.rb]: This route is incorrect now.

**Comment by namusyaka** [comment]
@dblock I think we should add the explanation about this into UPGRADING.md, right?

File: lib/grape/router.rb [hunk 2]
```diff
@@ -19,6 +19,10 @@ def self.normalize_path(path)
       path
     end
 
+    def self.supported_methods
+      @supported_methods ||= Grape::Http::Headers::SUPPORTED_METHODS + ['ANY']
```
Comment (dblock) [inline lib/grape/router.rb]: Maybe `*` instead of `ANY`? Just in case `ANY` one day becomes an HTTP verb :)

**Comment by dblock** [comment]
Definitely needs a note in UPGRADING. Thanks!

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ Next Release
 #### Fixes
 
 * [#1505](https://github.com/ruby-grape/grape/pull/1505): Run `before` and `after` callbacks, but skip the rest when handling OPTIONS - [@jlfaber](https://github.com/jlfaber).
+* [#1517](https://github.com/ruby-grape/grape/pull/1517): Modify the priority of ANY routes, fixes #1089 - [@namusyaka](https://github.com/namusyaka), [@wagenet](https://github.com/wagenet)
```
Comment (namusyaka) [inline CHANGELOG.md]: omg, thanks :)

File: lib/grape/router.rb (hunk 2 above)
Comment (namusyaka) [inline lib/grape/router.rb]: Got it.

**Comment by namusyaka** [comment]
@dblock Could you review this again?

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ Next Release
 #### Fixes
 
 * [#1505](https://github.com/ruby-grape/grape/pull/1505): Run `before` and `after` callbacks, but skip the rest when handling OPTIONS - [@jlfaber](https://github.com/jlfaber).
+* [#1517](https://github.com/ruby-grape/grape/pull/1517): Modify the priority of ANY routes, fixes #1089 - [@namusyaka](https://github.com/namusyaka), [@wagenet](https://github.com/wagenet).
```
Comment (dblock) [inline CHANGELOG.md]: Should be:

```
 [#1517](https://github.com/ruby-grape/grape/pull/1517), [#1089](https://github.com/ruby-grape/grape/pull/1089): Fix: priority of ANY routes - [@namusyaka](https://github.com/namusyaka), [@wagenet](https://github.com/wagenet).
```

Comment (namusyaka) [inline CHANGELOG.md]: Okay, fixed.

**Comment by dblock** [comment]
I corrected some English and merged via https://github.com/ruby-grape/grape/commit/e489995e79f0f426990b6934451977a7ecc462cd. Thank you.

**Comment by namusyaka** [comment]
@dblock Thanks!

File: spec/grape/api_spec.rb [hunk 5]
```diff
@@ -508,6 +508,37 @@ class DummyFormatClass
...
+        get 'something' do
+          'something'
+        end
+
+        route :any, '*path' do
+          'catch-all'
+        end
+      end
+
+      send('get', 'nested')
```
Comment (namusyaka) [inline spec/grape/api_spec.rb]: Right. I missed this. Sorry ;(

**Pull request closed without merging**